protocol/src/main/java/org/geysermc/mcprotocollib/protocol/ClientListener.java

```diff
@@ -91,13 +93,14 @@ public void packetReceived(Session session, Packet packet) {
                 }
 
                 session.send(new ServerboundKeyPacket(helloPacket.getPublicKey(), key, helloPacket.getChallenge()));
-                session.enableEncryption(protocol.enableEncryption(key));
+                session.setEncryption(protocol.enableEncryption(key));
             } else if (packet instanceof ClientboundGameProfilePacket) {
                 session.send(new ServerboundLoginAcknowledgedPacket());
             } else if (packet instanceof ClientboundLoginDisconnectPacket loginDisconnectPacket) {
                 session.disconnect(loginDisconnectPacket.getReason());
             } else if (packet instanceof ClientboundLoginCompressionPacket loginCompressionPacket) {
-                session.setCompressionThreshold(loginCompressionPacket.getThreshold(), false);
+                session.setCompression(loginCompressionPacket.getThreshold() >= 0 ?
+                    new CompressionConfig(loginCompressionPacket.getThreshold(), new ZlibCompression(), false) : null);
```

Comment:
save threshold to a variable just to make it more readable

Reply:
done